src/components/NowPlaying.tsx

```diff
@@ -112,62 +158,264 @@ class NowPlaying extends React.Component<{}, {audioFeatures: AudioFeaturesRespon
             {/* Recommendation #1 */}
             <div className={styles.trackContainer}>
               {/* Recommendation cover */}
-              <img className={styles.recommendationsCover} src={"https://upload.wikimedia.org/wikipedia/en/d/dd/Ray_of_Light_Madonna.png"}/>
+              <img className={styles.recommendationsCover} src={Object.keys(this.state.recommendations).length > 0 ? 
+                                                                  (this.state.recommendations as GetRecommendationsResponse)["tracks"][0].album.images[0].url : 
+                                                                ""}/>
               {/* Recommendation track details */}
               <div className={styles.trackDetils}>
-                <div className={styles.trackName}>{"Ray of light"}</div>
-                <div>{"Madonna"}</div>
-                <div>{"Ray of light"}</div>
+                <div className={styles.trackName}>
+                  {Object.keys(this.state.recommendations).length > 0 ? 
+                    (this.state.recommendations as GetRecommendationsResponse)["tracks"][0].name : 
+                   ""}
+                </div>
+                  {Object.keys(this.state.recommendations).length > 0 ? ( () => {
+                    // Get all the artists
+                    const trackArtists = (this.state.recommendations as GetRecommendationsResponse)["tracks"][0].artists;
+                    let trackAritistsInnnerHTML = "";
+
+                    // Check if there are any artists
+                    if (trackArtists) {
+                      // Display all the artists
+                      for (const artist of trackArtists) {
+                        trackAritistsInnnerHTML += (artist.name + ", ")
+                      }
+                      if(trackAritistsInnnerHTML.length > 0) {
+                        trackAritistsInnnerHTML = trackAritistsInnnerHTML.substring(0, trackAritistsInnnerHTML.length - 2);
+                      }
+
+                      return (<div className={styles.text}> 
+                                {trackAritistsInnnerHTML} 
+                              </div>);
+                    } else {
+                      return <></>;
+                    }
+
+                    })() : <div></div>}
+                <div>
+                {Object.keys(this.state.recommendations).length > 0 ? 
+                    (this.state.recommendations as GetRecommendationsResponse)["tracks"][0].album.name : 
+                 ""}
+                </div>
```

Comment:
There should be a way to put this code in a loop rather than copy and pasting it 6 times right?

Also, we might want to handle the case where 6 recommendations cannot be generated since it's not guaranteed to generate as many recommendations as we ask for. But to be honest I'd imagine that case is pretty rare so I wouldn't say it's urgent, but is something to maybe consider.